core/src/main/java/com/linecorp/armeria/common/RequestContext.java

```diff
@@ -597,26 +597,18 @@ default ContextAwareBlockingTaskExecutor makeContextAware(BlockingTaskExecutor e
      * Returns a {@link Callable} that makes sure the current {@link RequestContext} is set and then invokes
      * the input {@code callable}.
      */
-    default <T> Callable<T> makeContextAware(Callable<T> callable) {
+    default <T> ContextAwareCallable<T> makeContextAware(Callable<T> callable) {
         requireNonNull(callable, "callable");
-        return () -> {
-            try (SafeCloseable ignored = push()) {
-                return callable.call();
-            }
-        };
+        return ContextAwareCallable.of(this, callable);
     }
 
     /**
      * Returns a {@link Runnable} that makes sure the current {@link RequestContext} is set and then invokes
      * the input {@code runnable}.
      */
-    default Runnable makeContextAware(Runnable runnable) {
+    default ContextAwareRunnable makeContextAware(Runnable runnable) {
```

Comment:
This PR itself looks okay but it would be better to make all other `RequestContext.makeContextAware()` methods return context-aware variants such as `ContextAwareFunction` for consistency. 

We may handle it in a separate topic.